src/librustc_codegen_ssa/mir/block.rs

```diff
@@ -528,6 +528,16 @@ impl<'a, 'tcx, Bx: BuilderMethods<'a, 'tcx>> FunctionCx<'a, 'tcx, Bx> {
             _ => FnType::new(&bx, sig, &extra_args)
         };
 
+        // This should never be reachable at runtime:
+        // We should only emit a call to this intrinsic in #[cfg(miri)] mode,
+        // which means that we will never actually use the generate object files
+        // (we will just be interpreting the MIR)
```

Comment:
Given this comment, why not `bug!` out here?

Reply:
It turns out that when we build `libstd` with `xargo` for Miri, we will actually attempt to codegen `miri_start_panic` *for LLVM*, since we're generating object files in addition to MIR. Those object files never get used, but we still need to do *something* here so that compilation succeeds.

Reply:
Ah, I see. Please explain that in a comment.

Reply:
Updated